XUtils.h

```diff
@@ -56,6 +56,17 @@ void String_freeArray(char** s);
 
 char* String_readLine(FILE* fd) ATTR_MALLOC;
 
+static inline char *String_charNull(const char* s, int c) {
```

Comment:
Nitpick: The character is called `NUL` in the ASCII standard.

Any objections to calling this wrapper `String_strchrnul`?

Reply:
```suggestion
static inline char *String_charNull(const char* s, int c) ATTR_NONNULL {
```

Reply:
OK, no objections - will push these changes shortly, thanks.  Hmm, looks like the latter is invalid - results in:

```
XUtils.h:59:1: error: attributes should be specified before the declarator in a function definition
   59 | static inline char *String_strchrnul(const char* s, int c) ATTR_NONNULL {
      | ^~~~~~
```

I'll drop that second suggestion as a result.  If it was feasible, String_eq and String_startsWith in this header file could use the same treatment so I guess its not possible.